util/types/datum.go

```diff
@@ -1131,6 +1131,11 @@ func ProduceDecWithSpecifiedTp(dec *MyDecimal, tp *FieldType, sc *variable.State
 			}
 		}
 	}
+
+	if terror.ErrorEqual(err, ErrOverflow) {
```

Comment:
@XuHuaiyu no, because the variable.session.go should not import "types" package, because "types" use "variable" package, it will cause circle import problem